jazz_environments/components/validation/validate_create_payload.js

```diff
@@ -0,0 +1,190 @@
+            //call services get
+            svcGetPayload = {
+                uri: global.config.SERVICE_API_URL + global.config.SERVICE_API_RESOURCE + "?domain=" + service_domain + "&service=" + service_name,
+                method: "GET",
+                headers: {'Authorization':global.authorization},
+                rejectUnauthorized: false
+            };
+            request(svcGetPayload, function(error, response, body) {
+                if (response.statusCode === 200) {
+                    var output = JSON.parse(body);
+                    if (output.data === null || output.data === "" || output.data === undefined || output.data.available === undefined) {
+                        onComplete({
+                            result: "inputError",
+                            message: "Error finding service: " + service_domain + "." + service_name + " in service catalog"
+                        });
+                    } else if (output.data.available === false) {
+                        onComplete(null, {
+                            result: "success",
+                            message: "Service is available!"
+                        });
+                    } else if (output.data.available === true) {
+                        onComplete({
+                            result: "inputError",
+                            message: "Service with domain: " + service_domain + " and service name:" + service_name + ", does not exist."
+                        });
+                    }
+                } else {
+                    onComplete({
+                        result: "inputError",
+                        message: "Error finding service: " + service_domain + "." + service_name + " in service catalog"
+                    });
+                }
+            });
+        },
+        validateEnvironmentExists: function(onComplete) {
+            // Check if a environment with same environment exists
+            var query;
+
+            environment_data.logical_id = environment_data.logical_id.toLowerCase();
+            environment_data.service = environment_data.service.toLowerCase();
+            environment_data.domain = environment_data.domain.toLowerCase();
+
+            query = { logical_id: environment_data.logical_id, service: environment_data.service, domain: environment_data.domain };
+            logger.info("validateEnvironmentExists: "+JSON.stringify(query));
+            crud.getList(query, envTableName, function onServiceGet(error, data) {
+                if (error) {
+                    onComplete(error, null);
+                } else {
+                    if (data.environment.length > 0) {
+                        onComplete({
+                            result: "inputError",
+                            message: "The specified environment already exists, please choose a different logical id for your new environment"
+                        });
+                    } else {
+                        if (
```

Comment:
use shortform `if (environment_data.physical_id `
                          `     && environment_data.logical_id !== config.service_environment_production_logical_id `
                          `     && environment_data.logical_id !== config.service_environment_stage_logical_id)`

Reply:
Worked on the changes